build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetSDKs.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Xamarin.Android.Prepare
+{
+	class Step_InstallDotNetSDKs : StepWithDownloadProgress
+	{
+		public Step_InstallDotNetSDKs ()
+			: base ("Install required .NET Core SDKs locally")
+		{ }
+
+		protected override async Task<bool> Execute (Context context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+
+			var dotnetPath = context.Properties.GetRequiredValue (KnownProperties.DotNetPath);
+			dotnetPath = dotnetPath.TrimEnd (new char [] { Path.DirectorySeparatorChar });
+			var dotnetVersion = Configurables.Defaults.DotNetVersion;
+			var dotnetPreviewVersion = Configurables.Defaults.DotNetPreviewVersion;
+
+			// Delete any custom Microsoft.Android packs that may have been installed by test runs. Other ref/runtime packs will be ignored.
+			var packsPath = Path.Combine (dotnetPath, "packs");
+			if (Directory.Exists (packsPath)) {
+				foreach (var packToRemove in Directory.EnumerateDirectories (packsPath).Where (p => new DirectoryInfo (p).Name.Contains ("Android")))
+					Utilities.DeleteDirectory (packToRemove);
+			}
+
+			// Delete any unnecessary SDKs if they exist.
+			var sdkPath = Path.Combine (dotnetPath, "sdk");
+			if (Directory.Exists (sdkPath)) {
+				var installedSdks = Directory.EnumerateDirectories (sdkPath).Select (d => new DirectoryInfo (d));
+				foreach (var sdkToRemove in installedSdks.Where (s => s.Name != dotnetVersion && s.Name != dotnetPreviewVersion))
+					Utilities.DeleteDirectory (sdkToRemove.FullName);
+			}
```

Comment:
I feel the deletion code should be moved till after the `InstallDotNetAsync` calls, so that we delete old dotnet only if the new was installed successfully, thus assuring that the system is left in the same state as before `xaprepare` was invoked

Reply:
@grendello this new sandbox is going to be primarily for testing, and modified regularly to drop in various Android workload components. I'd like to be regularly cleaning it up even if not installing a new version so that we try to reduce the chances of getting into a weird state.